readthedocs/doc_builder/environments.py

```diff
@@ -294,7 +297,7 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_value, tb):
         ret = self.handle_exception(exc_type, exc_value, tb)
-        self.build['state'] = BUILD_STATE_FINISHED
+        self.update_build(BUILD_STATE_FINISHED)
```

Comment:
Won't this post that status to the API? Seems like we don't want to publish here, but only below if `finalize` is `False`.

Reply:
The `finalize` check is in `update_build`, so it should only post if `finalize` is `True`

Reply:
Gah ok. Lost that in the review.